src/pylibs/db/sql_adapter.py

```diff
@@ -0,0 +1,130 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sqlalchemy import create_engine, MetaData, Table, ForeignKey, Column, String, Date, LargeBinary, Boolean
+import logging
+import random
+from datetime import datetime
+
+# timestamp format for birthdays
+TIMESTAMP_FORMAT = '%Y-%m-%d'
+
+
+class SqlAdapter:
+    def __init__(self, uri):
+        # connect to db
+        self.engine = create_engine(uri)
+        self.users_table = Table(
+            'users',
+            MetaData(self.engine),
+            Column('accountid', String, primary_key=True),
+            Column('username', String, unique=True, nullable=False),
+            Column('passhash', LargeBinary, nullable=False),
+            Column('firstname', String, nullable=False),
+            Column('lastname', String, nullable=False),
+            Column('birthday', Date, nullable=False),
+            Column('timezone', String, nullable=False),
+            Column('address', String, nullable=False),
+            Column('state', String, nullable=False),
+            Column('zip', String, nullable=False),
+            Column('ssn', String, nullable=False),
+        )
+        self.contacts_table = Table(
+            'contacts',
+            MetaData(self.engine),
+            Column('username', String, ForeignKey(self.users_table.c.username), nullable=False),
+            Column('label', String, nullable=False),
+            Column('account_num', String, nullable=False),
+            Column('routing_num', String, nullable=False),
+            Column('is_external', Boolean, nullable=False),
+        )
+
+    def add_user(self, user):
+        """Add a user to the database.
+
+        Params: user - a key/value dict of attributes describing a new user
+                    {'username': username, 'password': password, ...}
+        Raises: SQLAlchemyError if there was an issue with the database
+        """
+        statement = self.users_table.insert().values(user)
+        logging.debug('QUERY: %s', str(statement))
+        with self.engine.connect() as conn:
+            conn.execute(statement)
+
+    def generate_accountid(self):
+        """Generates a globally unique alphanumerical accountid."""
+        accountid = None
+        with self.engine.connect() as conn:
+            while accountid is None:
+                accountid = str(random.randint(1e9, (1e10 - 1)))
+
+                statement = self.users_table.select().where(self.users_table.c.accountid == accountid)
+                logging.debug('QUERY: %s', str(statement))
+                result = conn.execute(statement).first()
+                logging.debug('RESULT: %s', str(result))
+                # If there already exists an account, try again.
+                if result is not None:
+                    accountid = None
+        return accountid
+
+    def get_user(self, username):
```

Comment:
Is there a reason why both services need access to all these functions? The design philosophy behind microservices seems to be to [avoid creating shared libraries](https://medium.com/standard-bank/microservices-dont-create-shared-libraries-2e803b033552) whenever possible. Is there a strong argument for coupling these services so tightly?

Reply:
@Daniel-Sanche yeah our initial plan with common lib was for noSQL+SQL(MySQL, PGSQL etc) but since now we are only targeting SQL I think we avoid creating on a common lib as SQLAlchemy will handle all SQL DBs.

Reply:
Ok sounds good. Will you be updating this PR, or creating a new one?

Reply:
I will open a new PR